### PR TITLE
Adds originating_functions to copy_to

### PR DESCRIPTION
**Pull request by elijahbenizzy**
[Short description explaining the high-level reason for the pull request]

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Review by skrawcz — COMMENTED** [review]
unit tests?

File: tests/test_node.py [hunk 1]
```diff
@@ -20,3 +20,16 @@ def fn():
...
     with pytest.raises(ValueError):
         Node.from_fn(fn)
+
+
+def test_node_copy_with_retains_originating_functions():
+    def fn() -> int:
+        return 1
+
+    node = Node.from_fn(fn)
+    node_copy = node.copy_with(name="rename_fn", originating_functions=(fn,))
```
Comment (skrawcz) [inline tests/test_node.py:30]: You shouldn't need `originating_functions=(fn,)` here right?

**Review by skrawcz — CHANGES_REQUESTED** [review]
Um, I don't think this fixes it.

You need to actually populate it the first time.  E.g. [here](https://github.com/DAGWorks-Inc/hamilton/pull/208/commits/e726fe778aac16053186255a2cf7b3163d05e3dd#diff-6b4a15b6f5263150ab8674431da707b5ea1c4a855e07a42ffb79ff5aa56cea86R209-R215)

**Review by skrawcz — APPROVED** [review]
Ah okay I understand now. This is on purpose.

File: tests/test_node.py (hunk 1 above)
Comment (elijahbenizzy) [reply tests/test_node.py:30]: Nope its currently assigned in `resolve_nodes`

**Review by skrawcz — APPROVED** [review]

**Pull request merged**